plugin/latexlivepreview.vim

```diff
@@ -82,11 +83,38 @@ function! s:Compile()
     " Write the current buffer in a temporary file
     silent exec 'write! ' . b:livepreview_buf_data['tmp_src_file']
 
-    call s:RunInBackground(b:livepreview_buf_data['run_cmd'])
+    " I have to wait for the compilation to finish, so that the synctex file
+	" can be updated.
+	" BEFORE: vim did not wait for compilation.
+    "call s:RunInBackground(b:livepreview_buf_data['run_cmd'])
+
+    " Now: vim waites until compilation is finished to modify the synctex file
+    silent call system(b:livepreview_buf_data['run_cmd'])
+    if v:shell_error != 0
+        echo 'Failed to compile'
+        lcd -
+        return
+    endif
+
+	" FIXME: sometimes if a source.tex file is not opened, the local path
+	" inside vim is messed up. The following line restores the local path
+	" when recomilation is initiated.
+    execute 'lcd ' . b:livepreview_buf_data['root_dir']
+
+    " The following established the source file name in the synctex.gz file.
+    " unzip -> sed change tex-source in synctex -> zip
+    let l:tmp_synctex = b:livepreview_buf_data['tmp_src_file'].".synctex"
```

Comment:
Since the synctex is in the tmp dir, how should the pdf reader pick it up? I tried the change in my vim and I don't see it's in effect. (I'm using evince and it supports synctex).

Reply:
In my case, the zathura pdf-viewer uses the synctex file in the location where the pdf is. Therefore, I did not need to tell zathura where the file is located. I can configure evince as a previewer. However, I need to check how to get synctex working with evince. * So I need some time...